packages/extension-dapp/src/bundle.ts

```diff
@@ -60,10 +63,22 @@ function filterAccounts (list: InjectedAccount[], genesisHash?: string | null, t
 
 /** @internal retrieves all the extensions available on the window */
 function getWindowExtensions (originName: string): Promise<InjectedExtension[]> {
+  let extensions: Record<string, InjectedWindowProvider>;
+  
+  /** If originName indicates enabling only snap */
+  if (['onlysnap', 'only_snap', 'snaponly', 'snap_only'].includes(originName.toLowerCase())) {
+    extensions = { DEFAULT_SNAP_NAME : injectedMetamaskSnap }; // no other extensions
+  } else {
+    extensions = win.injectedWeb3;
+  }
+
+  /** inject snap into window */
+  hasMetamask && (win.injectedWeb3[DEFAULT_SNAP_NAME] = injectedMetamaskSnap);
```

Comment:
You can't just pick a default here, you should inject all snaps with some convention such as `metamask_snap_${snapName}`.

Reply:
We have utilized a snapList that will be injected instead of a default one.